tianshou/data/batch.py

```diff
@@ -534,21 +576,24 @@ def __len__(self) -> int:
         return min(r)
 
     @property
-    def size(self) -> int:
-        """Return self.size."""
+    def shape(self) -> List[int]:
+        """Return self.shape."""
         if len(self.__dict__.keys()) == 0:
-            return 0
+            return []
         else:
-            r = []
+            data_shape = []
             for v in self.__dict__.values():
-                if isinstance(v, Batch):
-                    r.append(v.size)
-                elif hasattr(v, '__len__') and (not isinstance(
-                        v, (np.ndarray, torch.Tensor)) or v.ndim > 0):
-                    r.append(len(v))
-                else:
-                    r.append(1)
-            return min(r) if len(r) > 0 else 0
+                try:
+                    if len(v.shape) > 0:
+                        data_shape.append(v.shape)
+                except AttributeError:
+                    if isinstance(v, list):
+                        data_shape.append(tuple([len(v)]))
+                    else:
+                        raise TypeError("No support for 'shape' method with "
+                                        f"type {type(v)} in class Batch.")
+            return list(map(min, zip(*data_shape))) if len(data_shape) > 1 \
+                else data_shape[0] if len(data_shape) > 0 else []
```

Comment:
No need for `data_shape[0] if len(data_shape) > 0 else []` since `len(data_shape)` cannot be 0.

Reply:
Nope.
```python
In [3]: batch2 = Batch(a=[{ 
   ...:         'b': np.float64(1.0), 
   ...:         'c': np.zeros(1), 
   ...:         'd': Batch(e=np.array(3.0))}])

In [6]: batch2[0]
Out[6]: 
Batch(
    a: Batch(
           d: Batch(
                  e: 3.0,
              ),
           c: array([0.]),
           b: 1.0,
       ),
)
```
where batch2[0].shape will cause data_shape == [].